### ISSUE TITLE
[Bug]: Cannot get WebGPU adapter in Linux.

### ISSUE THREAD

**Issue by ezorzin** [electron/electron#41763]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

29.1.6

### What operating system are you using?

Other Linux

### Operating System Version

Manjaro

### What arch are you using?

x64

### Expected Behavior

When asking for a webgpu adapter I do:
`const { app, BrowserWindow } = require('electron')
app.commandLine.appendSwitch('enable-unsafe-webgpu'); `
…
`adapter = await navigator.gpu.requestAdapter();`

I expect to have the webgpu adapter returned.

### Actual Behavior

In Windows or Mac the function returns the adapter, but in Linux I get “null”.
Moreover, on the same Linux machine where this happens I am able to run all webgpu examples available online by means of the Google Chrome browser by enabling the enable-unsafe-webgpu flag of it.
I believe this proves on my Linux machine webgpu through Electron should work.

### Additional Information

In my webgpu/electron tests I got inspired by [this example](https://github.com/xhcao/webgpu-pipeline-electron/tree/main) and [this issue](https://github.com/electron/electron/issues/26944).

**Comment by mlaurencin** [comment]
@ezorzin Did you use the example you linked for your testing, or did you use it as a reference? If the latter, can you link a standalone testcase with only the code necessary to reproduce the issue? For example, [Electron Fiddle](https://www.electronjs.org/fiddle) is a great tool for making small test cases and makes it easy to publish your test case to a [gist](https://gist.github.com) that Electron maintainers can use.

Stand-alone test cases make fixing issues go more smoothly: it ensure everyone's looking at the same issue, it removes all unnecessary variables from the equation, and it can also provide the basis for automated regression tests.

I've added `blocked/need-info` label for this reason. After you make a test case, please link to it in a follow-up comment.

**Comment by mlaurencin** [comment]
Ok, and does this test produce the same results when using Chrome?

**Comment by ezorzin** [comment]
Yes, for instance you can try opening the same gist files with VScode and using the [live server](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer) extension.
I did this on the same Mac computer:
![WebGPU_red_triangle_VScode_Mac](https://github.com/electron/electron/assets/15363242/48916d00-4ef6-4832-a3a5-78d9111938bd)

and on the same Linux computer:
![WebGPU_red_triangle_VScode_Linux](https://github.com/electron/electron/assets/15363242/42b0a558-e13f-4475-ab92-66963f220e14)

using Google Chrome on both computers.

However, on Linux I have to activate these flags in Google Chrome in order to have WebGPU (also the various examples one can find online on the web) working:
![Chrome_flags_Linux](https://github.com/electron/electron/assets/15363242/6a68c122-93bb-42f3-94e1-c32223b988e5)

If I do not do that, then no red triangle is displayed on Linux:
![Without_experimental_flags_Linux](https://github.com/electron/electron/assets/15363242/1108cfdf-864e-4889-a8ef-fd1f5c7488f2)

For this, I believe maybe Electron is somehow not correctly managing those flags when working on Linux.

**Comment by ezorzin** [comment]
P.S. in Fiddle, under Linux, I also tried running the same code with different versions of electron: v29.1.5, v29.1.6, v.29.2.0 and v30.0.0-beta6. Nothing changes: the red triangle is still not displayed in Linux through Electron.

**Comment by mlaurencin** [comment]
After reading through the issue you linked, it looks like this feature has now been enabled by default. In that case, you may be experiencing issues with another flag or setting instead. I am going to close this issue as this doesn't appear to be an issue pertaining to webgpu itself. I would encourage taking a look at [this issue](https://github.com/electron/electron/issues/40929) to see if you are experiencing something similar, and if not, giving [the Electron Discord server](https://discord.com/invite/electron) a try to ask how others have worked with this feature.

I hope this helps!

**Comment by ezorzin** [comment]
I am sorry, but this did not help.
Could you please consider to show what is wrong in the code I posted as gist?
If it is not a problem of webgpu itself, nor of electron, could you please provide an example of webgpu code implemented in electron that works also in Linux?

I think your early decision of closing this bug without any analysis of the code I provided, nor any counterexample, is not satisfactory.

A gist testable in Fiddle of a simple working example, similar to mine, that proves electron can work with webgpu in Linux would be ideal: I kindly ask you to consider reopening this bug until this situation is fully clarified.

Thanks

**Comment by ezorzin** [comment]
To be constructive, I tried to investigate myself. According to [this](https://warpdesign.fr/electron-development-how-to-configure-nodejs-and-runtime-chrome-flags) there exists an app.commandLine.appendSwitch method of electron to pass command line arguments to the underlying Chromium API. This should be good to pass the "--enable-unsafe-webgpu" flag I need, but I found [this docs](https://www.electronjs.org/docs/latest/api/command-line-switches) showing a list of supported switches: the "--enable-unsafe-webgpu" one and many others (e.g. like "--enable-vulkan") are not there.
If I pass those ones then they are simply ignore by electron.

There is no doubt they are needed in Linux, as I demonstrated in my previous comments.

How can I pass them through electron?

**Comment by blurymind** [comment]
Why was this closed? Web gpu still broken on electron on linux and there is no way to enable it like you can in chrome

here is how you get it on chrome
https://stackoverflow.com/questions/72294876/i-enable-webgpu-in-chrome-dev-and-it-still-doesnt-work/72495310

**Comment by blurymind** [comment]
I am assuming this might be solved on linux electron by doing this

```
import { app } from 'electron';
app.commandLine.appendSwitch('enable-features', "SharedArrayBuffer")
app.commandLine.appendSwitch('enable-unsafe-webgpu')
app.commandLine.appendSwitch('enable-features', "Vulkan")
```

Can someone confirm for me please?

I tested it on chrome 
https://popov72.github.io/OceanDemo/dist/index.html
https://github.com/jeremy-coleman/OceanDemo?tab=readme-ov-file

without the vulkan flag the demo loads, but is super slow. You need both flags
